packages/plugin-mcp/src/utils.ts

```
@@ -0,0 +1,68 @@
+import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio";
+import { StreamableHTTPClientTransport } from "@modelcontextprotocol/sdk/client/streamableHttp";
+
+import { ServerConfig, StdioConfig, StreamableHttpConfig } from "./types";
+
+export function clientCommand(
+  command: string | undefined,
+  serverScriptPath: string | undefined
+) {
+  let result;
+
+  if (command) {
+    result = command;
+  } else if (serverScriptPath) {
+    const isPython = serverScriptPath.endsWith(".py");
+    result = isPython
+      ? process.platform === "win32"
+        ? "python"
+        : "python3"
+      : process.execPath;
+  } else {
+    throw new Error(
+      "MCP config needs either {command,args} or serverScriptPath"
+    );
+  }
+
+  return result;
+}
+
+export function buildTransport(config: StdioConfig): StdioClientTransport;
+export function buildTransport(
+  config: StreamableHttpConfig
+): StreamableHTTPClientTransport;
+export function buildTransport(config: ServerConfig) {
+  let transport;
+  if ("url" in config) {
+    transport = new StreamableHTTPClientTransport(new URL(config.url));
+  } else {
```

the `else` is bogus, please remove and flatten the other conditional

---

actually not clear on this one specifically

---

apologies, since my preference is fail fast, you can flatten the if else by removing the else statement - this is a follow up on https://github.com/UraniumCorporation/maiar-ai/pull/151#discussion_r2072493519

---

Done. Let me know if any other change needs to happen there. A const variable declaration seems not needed as it would be the immediate next return so i just return instead